### PR TITLE
feat: add bot callback URL params (?cb=&ct=) for seamless plan review

### PR DESCRIPTION
**Pull request by aviadshiber**
## Summary

Adds support for bot-initiated plan review callbacks. When a bot (e.g. a Slack bot) generates a plan and posts the Plannotator URL, it can now embed callback parameters so the user never needs to copy-paste a URL back.

**URL format:**
```
https://share.plannotator.ai/?cb=<encoded_callback_url>&ct=<auth_token>#<plan_hash>
```

**Changes:**
- `packages/ui/hooks/useSharing.ts`: `getCallbackConfig()` reads `?cb=` and `?ct=` from both `window.location.search` and hash-embedded query params (param order doesn't matter)
- `packages/editor/App.tsx`: Toolbar shows **Approve Design** and **Send Feedback** buttons when callback config is present. Clicking sends a POST to the bot callback URL with the current annotated URL.
- `packages/ui/utils/callbackHttp.ts`: `executeCallback()` extracted as a pure testable function (no React deps).

**User flow:**
1. Bot generates plan → embeds `?cb=&ct=` into the Plannotator URL → posts to Slack
2. User opens URL → sees Approve/Feedback buttons in toolbar
3. User annotates → clicks **Send Feedback** → bot receives the annotated URL automatically
4. No copy-paste required

**Callback URL — not just localhost:**
The `?cb=` value is whatever HTTPS endpoint the bot exposes. Two supported modes:
- **Local dev**: `https://localhost:9456` — requires [mkcert](https://github.com/FiloSottile/mkcert) to avoid mixed-content blocking (bot auto-detects certs at `~/.config/plannotator-cb-certs/`)
- **K8s / deployed**: any public HTTPS ingress URL configured via `PLANNOTATOR_CALLBACK_BASE_URL` env var on the bot

**Error handling:**
- `401`: "Plan link expired — request a new one from the bot"
- Network error: "Callback failed"

**Security:**
- Buttons only render when `?cb=` and `?ct=` are present in the URL
- Token is single-use (server-side enforcement by the bot)
- Mixed-content blocking ensures `?cb=` must be HTTPS when opened from share.plannotator.ai

## Test plan
- [x] Open a URL without `?cb=&ct=` params → no callback buttons visible *(unit: `getCallbackConfig` returns `None` → `isSome` guard hides buttons)*
- [x] Open a URL with params before `#`: `?cb=https://...&ct=<token>#<hash>` → buttons appear *(unit: `getCallbackConfig` returns `Some`)*
- [x] Open a URL with params after `#`: `#<hash>?cb=https://...&ct=<token>` → buttons still appear *(unit: hash fallback parser)*
- [x] Click "Send Feedback" → POST sent with `action: "feedback"` and current `window.location.href` *(unit: `executeCallback` body assertion)*
- [x] Click "Approve Design" → POST sent with `action: "approve"` *(unit: `executeCallback` body assertion)*
- [x] Test 401 response → user sees "Plan link expired" message *(unit: `executeCallback` 401 case)*
- [x] Test network failure → user sees "Callback failed." message *(unit: `executeCallback` throw case)*

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by backnotprop** [comment]
Interesting! May be related to some of work im enabling in #400

**Comment by backnotprop** [comment]
@aviadshiber Im going to merge this.

It would be cool to know how youll use this / your use case. I would like to try

**Comment by aviadshiber** [comment]
Thanks  @backnotprop ! Happy to share the use case.

I run a Slack bot built on [Kapsis](https://github.com/aviadshiber/kapsis) — an AI agent sandbox that spins up Claude in isolated containers for engineering tasks. When an agent produces a design plan, the bot compresses it to a Plannotator URL and posts it to Slack with action buttons. With this PR, the bot now embeds `?cb=&ct=` into that URL so the reviewer can annotate the plan in Plannotator and click "Submit Feedback" directly — the callback POSTs back to the bot, which re-runs the agent with the annotations. No copy-pasting URLs back into Slack.

Also — genuinely love Plannotator, I use it all the time. Great product, thanks for building it! Looking forward to seeing what you're enabling in #400!

**Comment by aviadshiber** [comment]
### Fixes made

**`packages/ui/hooks/useSharing.ts`**
- `getCallbackConfig()` now accepts an optional `loc` parameter typed as `{ readonly search: string; readonly hash: string }` (default: `window.location`). This makes the function unit-testable without a DOM environment — no `happy-dom`, no `@happy-dom/global-registrator`, no `Object.defineProperty(window, ...)` needed.

**`packages/ui/hooks/useSharing.test.ts`**
- Replaced the `setLocation()` helper (which used `Object.defineProperty(window, "location", ...)`) with a pure `loc(url)` factory that parses a URL string and returns `{ search, hash }`. This removes all DOM globals from the test file.
- All 6 original tests ported and passing.
- Added 3 new test cases:
  - `returns None when params are empty strings` — guards against `?cb=&ct=` (empty values should not produce a config)
  - `partial params in hash — only hash cb, no ct` — confirms partial hash params are still rejected
  - `K8s-style URL with encoded colons and slashes decodes correctly` — covers `http://plannotator-cb.svc.cluster.local:9456/callback` style callback URLs

**Test result:** 9/9 pass with `bun test` — no DOM setup required.

### No changes needed in `App.tsx`
The current branch already uses `isSome(callbackConfig)` guards (not `!== null`) and the `setTimeout(..., 4000)` is correctly placed after the `try/catch` block.

**Comment by aviadshiber** [comment]
Pushed follow-up refactor:
- **Option monad**: `getCallbackConfig()` now returns `Option<CallbackConfig>` (new `packages/ui/utils/option.ts`, zero deps) instead of `T | null`
- **Tests added**: `packages/ui/hooks/useSharing.test.ts` with bun:test — covers None (no params), Some before `#`, Some after `#` (param order independence), partial params, encoded URL decoding

**Comment by aviadshiber** [comment]
pushed several improvements before you merge:

- **Tests**: 16 tests added for `getCallbackConfig` (param parsing, hash-embedded params, K8s URLs, edge cases) and `executeCallback` (200/401/500/network error, request body assertions) — zero DOM dependencies
- **Non-localhost**: Updated description to clarify `?cb=` accepts any HTTPS URL — K8s ingress, mkcert local dev, or public endpoint
- **Code cleanup**: Consolidated callback logic into a single `utils/callback.ts` (no ad-hoc modules), aligned button styling to the existing Approve/Send Feedback look, using `T | null` consistent with the rest of the codebase

File: packages/ui/utils/callback.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    });
+    if (!res.ok) {
+      return {
+        type: "error",
+        message: res.status === 401
+          ? "Plan link expired — request a new one from the bot."
+          : "Callback failed.",
+      };
+    }
+    return { type: "success", message: successMsg };
```
Comment (aviadshiber) [inline packages/ui/utils/callback.ts]: will better have dedicated type for this

**Review by aviadshiber — COMMENTED** [review]
Review from Plannotator

**Comment by aviadshiber** [comment]
@backnotprop addressed the feedback — named `ToastSuccess`/`ToastError` types, tests, and code cleanup are all in. Ready to merge whenever you are, unless you have more feedback! 🙏

**Comment by backnotprop** [comment]
nice work @aviadshiber

**Comment by backnotprop** [comment]
@aviadshiber

We're building **Plannotator Workspaces**, where agent plans, specs, and reviews can be collaborated on and distilled into standards and rules. Your annotations and reviews compound into a golden ruleset planning and review agents can use, and CI/CD can verify against.

Joining the waitlist gets you lifetime free access as an early contributor. https://plannotator.ai/workspaces/. Check the OSS contributor box.

The OSS core isn't changing: same free and open source plugin, always.

Thanks for [#416](https://github.com/backnotprop/plannotator/pull/416).
